### PR TITLE
@Index annotation and some configuration tests are implemented

### PR DESCRIPTION
**Pull request by viktor44**
Creation additional indexes with @Index annotation

### CODE REVIEW

File: ignite/src/main/java/org/hibernate/ogm/datastore/ignite/impl/IgniteCacheInitializer.java [hunk 1]
```diff
@@ -149,6 +154,8 @@ private void initializeIdSources(SchemaDefinitionContext context, IgniteDatastor
 	private CacheConfiguration createCacheConfiguration(IdSourceKeyMetadata idSourceKeyMetadata) {
 		CacheConfiguration result = new CacheConfiguration();
 		result.setName( StringHelper.stringBeforePoint( idSourceKeyMetadata.getName() ) );
+		result.setAtomicityMode( CacheAtomicityMode.TRANSACTIONAL );
+		result.setCacheMode( CacheMode.REPLICATED );
```
Comment (schernolyas) [inline ignite/src/main/java/org/hibernate/ogm/datastore/ignite/impl/IgniteCacheInitializer.java:158]: it is unacceptable! 
Why you configure default CacheMode

File: ignite/src/main/java/org/hibernate/ogm/datastore/ignite/impl/IgniteCacheInitializer.java [hunk 2]
```diff
@@ -161,18 +168,25 @@ private CacheConfiguration createCacheConfiguration(AssociationKeyMetadata assoc
 		CacheConfiguration result = new CacheConfiguration();
 		result.setName( StringHelper.stringBeforePoint( associationKeyMetadata.getTable() ) );
 		result.setQueryEntities( Arrays.asList( queryEntity ) );
+		result.setCacheMode( CacheMode.PARTITIONED );
```
Comment (schernolyas) [inline ignite/src/main/java/org/hibernate/ogm/datastore/ignite/impl/IgniteCacheInitializer.java:171]: it is unacceptable! 
Why you configure default CacheMode ?

File: ignite/src/main/java/org/hibernate/ogm/datastore/ignite/impl/IgniteCacheInitializer.java [hunk 3]
```diff
@@ -185,9 +199,9 @@ private Class getEntityIdClassName( String table, SchemaDefinitionContext contex
 		CacheConfiguration<?,?> cacheConfiguration = new CacheConfiguration<>();
 		cacheConfiguration.setStoreKeepBinary( true );
 		cacheConfiguration.setSqlSchema( QueryUtils.DFLT_SCHEMA );
-		cacheConfiguration.setBackups( 1 );
 		cacheConfiguration.setName( StringHelper.stringBeforePoint( entityKeyMetadata.getTable() ) );
 		cacheConfiguration.setAtomicityMode( CacheAtomicityMode.TRANSACTIONAL );
+		cacheConfiguration.setCacheMode( CacheMode.PARTITIONED );
```
Comment (schernolyas) [inline ignite/src/main/java/org/hibernate/ogm/datastore/ignite/impl/IgniteCacheInitializer.java:204]: it is unacceptable! 
Why you configure default CacheMode ?

**Review by schernolyas — CHANGES_REQUESTED** [review]

**Comment by schernolyas** [comment]
I think the PR is for repository 'https://github.com/hibernate/hibernate-ogm-ignite'. Am I right?

File: ignite/src/main/java/org/hibernate/ogm/datastore/ignite/impl/IgniteCacheInitializer.java (hunk 1 above)
Comment (schernolyas) [reply ignite/src/main/java/org/hibernate/ogm/datastore/ignite/impl/IgniteCacheInitializer.java:158]: Could you please remove it to configuration if you need configure CacheMode.
And ... What will with other CacheMode?

Comment (viktor44) [reply ignite/src/main/java/org/hibernate/ogm/datastore/ignite/impl/IgniteCacheInitializer.java:158]: Why "it is unacceptable"? I think that replicated mode is more suitable for ID cache than partitioned

File: ignite/src/main/java/org/hibernate/ogm/datastore/ignite/impl/IgniteCacheInitializer.java (hunk 3 above)
Comment (viktor44) [reply ignite/src/main/java/org/hibernate/ogm/datastore/ignite/impl/IgniteCacheInitializer.java:204]: Why not? PARTITIONED is the default mode. Do you think that REPLICATED is better for entity cache by default?

Comment (schernolyas) [reply ignite/src/main/java/org/hibernate/ogm/datastore/ignite/impl/IgniteCacheInitializer.java:204]: Hi @Z-z-z-z !

I think property file is  the best way for configure cache mode of each cache. Your currect code blocks using other cache modes. And this is unacceptable. In this way, please, could you remove  cache mode configuration from code to property file?
Thanks!

**Comment by DavideD** [comment]
I cherry picked the changes and applied them on master.

I skipped the configuration changes, mainly because I don't know exactly what they do, maybe you can send another PR that only change that and add some description? Keep in mind that I'm not an expert w
with Apache Ignite, so you might need to describe something that it is obvious for you.

I'm a bit suspicious about using those defaults in `IgniteCacheInitializer`, shouldn't the user be able to configure them somehow?

The tests work without them anyway. 
We can create a separate issue for this one and discuss a bit more

**Pull request closed without merging**